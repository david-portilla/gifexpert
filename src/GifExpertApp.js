import React, { useState } from 'react'
import { AddCategory } from "./components/AddCategory";
import { GifList } from './components/GifList';

export const GifExpertApp = ({ defaultCategories = [] }) => {

  const [ categories, setCategories ] = useState(defaultCategories)

  return (
    <>
      <h2> GifExpert App</h2>
      <AddCategory setCategories={ setCategories } />
      <hr />
      <ol>
        {
          categories.map(category =>
            <GifList key={ category } category={ category } />
          )
        }
      </ol>
    </>
  )
}

export default GifExpertApp

