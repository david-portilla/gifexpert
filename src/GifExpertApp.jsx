import React, { useState } from "react";
import { SearchForm } from "./components/SearchForm";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [searchs, setSearchs] = useState([]);
  const onSearchTerm = (searchTerm) => {
    if (searchs.includes(searchTerm)) return;
    setSearchs([searchTerm]);
  };

  return (
    <>
      <h1>Get images from GIPHY</h1>
      <SearchForm onSearchTerm={onSearchTerm} />
      {searchs.map((search) => (
        <GifGrid key={search} term={search} />
      ))}
    </>
  );
};
