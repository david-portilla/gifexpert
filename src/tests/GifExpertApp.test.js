import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp';

describe('Test GifExpertApp ', () => {

  test('should matcth snapshoot', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should show a list of categories', () => {
    const categories = [ 'vegeta', 'goku' ]
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifList').length).toBe(categories.length)
  })

})
