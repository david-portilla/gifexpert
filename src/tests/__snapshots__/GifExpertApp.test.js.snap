// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExpertApp  should matcth snapshoot 1`] = `
<Fragment>
  <h2>
     GifExpert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test GifExpertApp  should show a list of categories 1`] = `
<Fragment>
  <h2>
     GifExpert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifList
      category="vegeta"
      key="vegeta"
    />
    <GifList
      category="goku"
      key="goku"
    />
  </ol>
</Fragment>
`;
